daemon/volumes.go

```diff
@@ -80,7 +81,7 @@ func (m *Mount) initialize() error {
 	if hostPath, exists := m.container.Volumes[m.MountToPath]; exists {
 		// If this is a bind-mount/volumes-from, maybe it was passed in at start instead of create
 		// We need to make sure bind-mounts/volumes-from passed on start can override existing ones.
-		if !m.volume.IsBindMount && m.from == nil {
+		if (!m.volume.IsBindMount && !m.isBind) && m.from == nil {
```

Comment:
Because the issue this is fixing isn't with volumes-from, it's doing something like `docker run -v /var/lib/docker/vfs/dir/55a6f3a06952bf1c6f8a93a85c55493107a6cbe8191a8d9e83ff75fad705e268:/foo ...`